From 335f2b596707070100db911606070b5b96b00554 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Tue, 10 Jun 2025 15:31:21 +0200
Subject: [PATCH 1/3] add: rhel 9.6, rhel 9.7 (prerelease)

Update rhel-9-unknown to match 9.8 and later.

f566665d rhel: Add RHEL 9.7 (prerelease)
27b4355f Add rhel 9.6 release date
5a0c2bab rhel: Add RHEL 9.6 (prerelease)

Resolves: RHEL-87787
Signed-off-by: rpm-build <rpm-build>
---
 os/redhat.com/rhel-9-unknown.xml |  28 ++---
 os/redhat.com/rhel-9.6.xml       | 170 +++++++++++++++++++++++++++++++
 os/redhat.com/rhel-9.7.xml       | 170 +++++++++++++++++++++++++++++++
 3 files changed, 354 insertions(+), 14 deletions(-)
 create mode 100644 os/redhat.com/rhel-9.6.xml
 create mode 100644 os/redhat.com/rhel-9.7.xml

diff --git a/os/redhat.com/rhel-9-unknown.xml b/os/redhat.com/rhel-9-unknown.xml
index f4ce9b7..20a0953 100644
--- a/os/redhat.com/rhel-9-unknown.xml
+++ b/os/redhat.com/rhel-9-unknown.xml
@@ -5,12 +5,12 @@
   <os id="http://redhat.com/rhel/9-unknown">
     <short-id>rhel9-unknown</short-id>
     <name>Red Hat Enterprise Linux 9 Unknown</name>
-    <name xml:lang="zh_Hans">Red Hat Enterprise Linux 9 Unknown</name>
+    <name xml:lang="zh-Hans">Red Hat Enterprise Linux 9 Unknown</name>
     <name xml:lang="ka">Red Hat Enterprise Linux 9 Unknown</name>
     <name xml:lang="ko">Red Hat &#xAE30;&#xC5C5;&#xC6A9; &#xB9AC;&#xB205;&#xC2A4; 9 (&#xC54C; &#xC218; &#xC5C6;&#xC74C;)</name>
     <name xml:lang="uk">Red Hat Enterprise Linux 9 &#x43D;&#x435;&#x432;&#x456;&#x434;&#x43E;&#x43C;&#x430;</name>
     <name xml:lang="tr">Red Hat Enterprise Linux 9 Bilinmeyen</name>
-    <name xml:lang="pt_BR">Red Hat Enterprise Linux 9 desconhecido</name>
+    <name xml:lang="pt-BR">Red Hat Enterprise Linux 9 desconhecido</name>
     <name xml:lang="pl">Red Hat Enterprise Linux 9 (nieznany)</name>
     <name xml:lang="it">Red Hat Enterprise Linux 9 (sconosciuto)</name>
     <name xml:lang="id">Red Hat Enterprise Linux 9 Tak Diketahui</name>
@@ -18,13 +18,13 @@
     <name xml:lang="de">Red Hat Enterprise Linux 9 Unknown</name>
     <version>9-unknown</version>
     <vendor>Red Hat, Inc</vendor>
-    <vendor xml:lang="zh_Hans">Red Hat, Inc</vendor>
+    <vendor xml:lang="zh-Hans">Red Hat, Inc</vendor>
     <vendor xml:lang="ka">Red Hat, Inc</vendor>
     <vendor xml:lang="ko">Red Hat, Inc</vendor>
     <vendor xml:lang="fi">Red Hat, Inc</vendor>
     <vendor xml:lang="uk">Red Hat, Inc</vendor>
     <vendor xml:lang="tr">Red Hat, Inc</vendor>
-    <vendor xml:lang="pt_BR">Red Hat, Inc</vendor>
+    <vendor xml:lang="pt-BR">Red Hat, Inc</vendor>
     <vendor xml:lang="pl">Red Hat, Inc</vendor>
     <vendor xml:lang="ja">Red Hat, Inc</vendor>
     <vendor xml:lang="it">Red Hat, Inc</vendor>
@@ -36,59 +36,59 @@
     <family>linux</family>
     <distro>rhel</distro>
     <codename>Plow</codename>
-    <upgrades id="http://redhat.com/rhel/9.4"/>
-    <derives-from id="http://redhat.com/rhel/9.4"/>
+    <upgrades id="http://redhat.com/rhel/9.7"/>
+    <derives-from id="http://redhat.com/rhel/9.7"/>
     <media arch="x86_64">
       <iso>
-        <volume-id>.*RHEL-9-([6-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?.*x86_64.*</volume-id>
+        <volume-id>.*RHEL-9-([8-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?.*x86_64.*</volume-id>
       </iso>
       <kernel>isolinux/vmlinuz</kernel>
       <initrd>isolinux/initrd.img</initrd>
     </media>
     <media arch="aarch64">
       <iso>
-        <volume-id>RHEL-9-([6-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?.*aarch64$</volume-id>
+        <volume-id>RHEL-9-([8-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?.*aarch64$</volume-id>
       </iso>
       <kernel>images/pxeboot/vmlinuz</kernel>
       <initrd>images/pxeboot/initrd.img</initrd>
     </media>
     <media arch="ppc64le">
       <iso>
-        <volume-id>RHEL[_-]9[_-]([6-9]|[1-9][0-9])([_-]([0-9]|[1-9][0-9]))?.*ppc64le$</volume-id>
+        <volume-id>RHEL[_-]9[_-]([8-9]|[1-9][0-9])([_-]([0-9]|[1-9][0-9]))?.*ppc64le$</volume-id>
       </iso>
       <kernel>ppc/ppc64/vmlinuz</kernel>
       <initrd>ppc/ppc64/initrd.img</initrd>
     </media>
     <media arch="s390x">
       <iso>
-        <volume-id>RHEL-9-([6-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?.*s390x$</volume-id>
+        <volume-id>RHEL-9-([8-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?.*s390x$</volume-id>
       </iso>
     </media>
     <tree arch="aarch64">
       <treeinfo>
         <family>Red Hat Enterprise Linux</family>
-        <version>^9.([6-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?$</version>
+        <version>^9.([8-9]|[1-9][0-9])(-[0-9]|[1-9][0-9])?$</version>
         <arch>aarch64</arch>
       </treeinfo>
     </tree>
     <tree arch="ppc64le">
       <treeinfo>
         <family>Red Hat Enterprise Linux</family>
-        <version>^9.([6-9]|[1-9][0-9])(.([0-9]|[1-9][0-9]))?$</version>
+        <version>^9.([8-9]|[1-9][0-9])(.([0-9]|[1-9][0-9]))?$</version>
         <arch>ppc64le</arch>
       </treeinfo>
     </tree>
     <tree arch="s390x">
       <treeinfo>
         <family>Red Hat Enterprise Linux</family>
-        <version>^9.([6-9]|[1-9][0-9])(.([0-9]|[1-9][0-9]))?$</version>
+        <version>^9.([8-9]|[1-9][0-9])(.([0-9]|[1-9][0-9]))?$</version>
         <arch>s390x</arch>
       </treeinfo>
     </tree>
     <tree arch="x86_64">
       <treeinfo>
         <family>Red Hat Enterprise Linux</family>
-        <version>^9.([6-9]|[1-9][0-9])(.([0-9]|[1-9][0-9]))?$</version>
+        <version>^9.([8-9]|[1-9][0-9])(.([0-9]|[1-9][0-9]))?$</version>
         <arch>x86_64</arch>
       </treeinfo>
     </tree>
diff --git a/os/redhat.com/rhel-9.6.xml b/os/redhat.com/rhel-9.6.xml
new file mode 100644
index 0000000..bf12c20
--- /dev/null
+++ b/os/redhat.com/rhel-9.6.xml
@@ -0,0 +1,170 @@
+<?xml version="1.0"?>
+<libosinfo version="0.0.1">
+  <!-- Licensed under the GNU General Public License version 2 or later.
+     See http://www.gnu.org/licenses/ for a copy of the license text -->
+  <os id="http://redhat.com/rhel/9.6">
+    <short-id>rhel9.6</short-id>
+    <name>Red Hat Enterprise Linux 9.6</name>
+    <version>9.6</version>
+    <vendor>Red Hat, Inc</vendor>
+    <vendor xml:lang="zh-Hans">Red Hat, Inc</vendor>
+    <vendor xml:lang="ka">Red Hat, Inc</vendor>
+    <vendor xml:lang="ko">Red Hat, Inc</vendor>
+    <vendor xml:lang="fi">Red Hat, Inc</vendor>
+    <vendor xml:lang="uk">Red Hat, Inc</vendor>
+    <vendor xml:lang="tr">Red Hat, Inc</vendor>
+    <vendor xml:lang="pt-BR">Red Hat, Inc</vendor>
+    <vendor xml:lang="pl">Red Hat, Inc</vendor>
+    <vendor xml:lang="ja">Red Hat, Inc</vendor>
+    <vendor xml:lang="it">Red Hat, Inc</vendor>
+    <vendor xml:lang="id">Red Hat, Inc</vendor>
+    <vendor xml:lang="fr">Red Hat, Inc</vendor>
+    <vendor xml:lang="es">Red Hat, Inc</vendor>
+    <vendor xml:lang="de">Red Hat, Inc</vendor>
+    <vendor xml:lang="ca">Red Hat, Inc</vendor>
+    <family>linux</family>
+    <distro>rhel</distro>
+    <codename>Plow</codename>
+    <upgrades id="http://redhat.com/rhel/9.5"/>
+    <derives-from id="http://redhat.com/rhel/9.5"/>
+    <release-date>2025-05-16</release-date>
+    <!-- RHEL 9.6 entries have to cope both with 9.6 and 9.6.x regexes for
+         medias' volume-id and trees' version -->
+    <media arch="x86_64">
+      <iso>
+        <volume-id>RHEL-9-6-(([0-9]|[1-9][0-9])-)?BaseOS-x86_64</volume-id>
+      </iso>
+      <kernel>isolinux/vmlinuz</kernel>
+      <initrd>isolinux/initrd.img</initrd>
+    </media>
+    <media arch="ppc64le">
+      <iso>
+        <volume-id>RHEL[_-]9[_-]6[_-](([0-9]|[1-9][0-9])[_-])?BaseOS[_-]ppc64le</volume-id>
+      </iso>
+      <kernel>ppc/ppc64/vmlinuz</kernel>
+      <initrd>ppc/ppc64/initrd.img</initrd>
+    </media>
+    <media arch="aarch64">
+      <iso>
+        <volume-id>RHEL-9-6-(([0-9]|[1-9][0-9])-)?BaseOS-aarch64</volume-id>
+      </iso>
+      <kernel>images/pxeboot/vmlinuz</kernel>
+      <initrd>images/pxeboot/initrd.img</initrd>
+    </media>
+    <media arch="s390x">
+      <iso>
+        <volume-id>RHEL-9-6-(([0-9]|[1-9][0-9])-)?BaseOS-s390x</volume-id>
+      </iso>
+    </media>
+    <tree arch="aarch64">
+      <treeinfo>
+        <family>Red Hat Enterprise Linux</family>
+        <version>^9.6(.([0-9]|[1-9][0-9]))?$</version>
+        <arch>aarch64</arch>
+      </treeinfo>
+    </tree>
+    <tree arch="ppc64le">
+      <treeinfo>
+        <family>Red Hat Enterprise Linux</family>
+        <version>^9.6(.([0-9]|[1-9][0-9]))?$</version>
+        <arch>ppc64le</arch>
+      </treeinfo>
+    </tree>
+    <tree arch="s390x">
+      <treeinfo>
+        <family>Red Hat Enterprise Linux</family>
+        <version>^9.6(.([0-9]|[1-9][0-9]))?$</version>
+        <arch>s390x</arch>
+      </treeinfo>
+    </tree>
+    <tree arch="x86_64">
+      <treeinfo>
+        <family>Red Hat Enterprise Linux</family>
+        <version>^9.6(.([0-9]|[1-9][0-9]))?$</version>
+        <arch>x86_64</arch>
+      </treeinfo>
+    </tree>
+    <resources arch="ppc64le">
+      <minimum>
+        <n-cpus>1</n-cpus>
+        <cpu>1000000000</cpu>
+        <ram>2147483648</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <cpu>4000000000</cpu>
+        <ram>2147483648</ram>
+        <storage>21474836480</storage>
+      </recommended>
+      <maximum>
+        <n-cpus>768</n-cpus>
+        <ram>35184372088832</ram>
+      </maximum>
+    </resources>
+    <resources arch="x86_64">
+      <minimum>
+        <cpu>1000000000</cpu>
+        <ram>1610612736</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <cpu>1000000000</cpu>
+        <ram>1610612736</ram>
+        <storage>21474836480</storage>
+      </recommended>
+      <maximum>
+        <n-cpus>768</n-cpus>
+        <ram>26388279066624</ram>
+      </maximum>
+      <network-install>
+        <ram>3221225472</ram>
+      </network-install>
+    </resources>
+    <resources arch="s390x">
+      <minimum>
+        <ram>1073741824</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <storage>21474836480</storage>
+      </recommended>
+      <maximum>
+        <n-cpus>340</n-cpus>
+        <ram>17592186044416</ram>
+      </maximum>
+      <network-install>
+        <ram>2147483648</ram>
+      </network-install>
+    </resources>
+    <resources arch="aarch64">
+      <minimum>
+        <ram>2147483648</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <storage>21474836480</storage>
+      </recommended>
+      <maximum>
+        <n-cpus>256</n-cpus>
+        <ram>1649267441664</ram>
+      </maximum>
+    </resources>
+    <resources arch="all">
+      <minimum>
+        <n-cpus>1</n-cpus>
+        <cpu>1000000000</cpu>
+        <ram>1610612736</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <cpu>1000000000</cpu>
+        <ram>1610612736</ram>
+        <storage>21474836480</storage>
+      </recommended>
+    </resources>
+    <installer>
+      <script id="http://redhat.com/rhel/kickstart/jeos"/>
+      <script id="http://redhat.com/rhel/kickstart/desktop"/>
+    </installer>
+  </os>
+</libosinfo>
diff --git a/os/redhat.com/rhel-9.7.xml b/os/redhat.com/rhel-9.7.xml
new file mode 100644
index 0000000..8f25a83
--- /dev/null
+++ b/os/redhat.com/rhel-9.7.xml
@@ -0,0 +1,170 @@
+<?xml version="1.0"?>
+<libosinfo version="0.0.1">
+  <!-- Licensed under the GNU General Public License version 2 or later.
+     See http://www.gnu.org/licenses/ for a copy of the license text -->
+  <os id="http://redhat.com/rhel/9.7">
+    <short-id>rhel9.7</short-id>
+    <name>Red Hat Enterprise Linux 9.7</name>
+    <version>9.7</version>
+    <vendor>Red Hat, Inc</vendor>
+    <vendor xml:lang="zh-Hans">Red Hat, Inc</vendor>
+    <vendor xml:lang="ka">Red Hat, Inc</vendor>
+    <vendor xml:lang="ko">Red Hat, Inc</vendor>
+    <vendor xml:lang="fi">Red Hat, Inc</vendor>
+    <vendor xml:lang="uk">Red Hat, Inc</vendor>
+    <vendor xml:lang="tr">Red Hat, Inc</vendor>
+    <vendor xml:lang="pt-BR">Red Hat, Inc</vendor>
+    <vendor xml:lang="pl">Red Hat, Inc</vendor>
+    <vendor xml:lang="ja">Red Hat, Inc</vendor>
+    <vendor xml:lang="it">Red Hat, Inc</vendor>
+    <vendor xml:lang="id">Red Hat, Inc</vendor>
+    <vendor xml:lang="fr">Red Hat, Inc</vendor>
+    <vendor xml:lang="es">Red Hat, Inc</vendor>
+    <vendor xml:lang="de">Red Hat, Inc</vendor>
+    <vendor xml:lang="ca">Red Hat, Inc</vendor>
+    <family>linux</family>
+    <distro>rhel</distro>
+    <codename>Plow</codename>
+    <upgrades id="http://redhat.com/rhel/9.6"/>
+    <derives-from id="http://redhat.com/rhel/9.6"/>
+    <release-status>prerelease</release-status>
+    <!-- RHEL 9.7 entries have to cope both with 9.7 and 9.7.x regexes for
+         medias' volume-id and trees' version -->
+    <media arch="x86_64">
+      <iso>
+        <volume-id>RHEL-9-7-(([0-9]|[1-9][0-9])-)?BaseOS-x86_64</volume-id>
+      </iso>
+      <kernel>isolinux/vmlinuz</kernel>
+      <initrd>isolinux/initrd.img</initrd>
+    </media>
+    <media arch="ppc64le">
+      <iso>
+        <volume-id>RHEL[_-]9[_-]7[_-](([0-9]|[1-9][0-9])[_-])?BaseOS[_-]ppc64le</volume-id>
+      </iso>
+      <kernel>ppc/ppc64/vmlinuz</kernel>
+      <initrd>ppc/ppc64/initrd.img</initrd>
+    </media>
+    <media arch="aarch64">
+      <iso>
+        <volume-id>RHEL-9-7-(([0-9]|[1-9][0-9])-)?BaseOS-aarch64</volume-id>
+      </iso>
+      <kernel>images/pxeboot/vmlinuz</kernel>
+      <initrd>images/pxeboot/initrd.img</initrd>
+    </media>
+    <media arch="s390x">
+      <iso>
+        <volume-id>RHEL-9-7-(([0-9]|[1-9][0-9])-)?BaseOS-s390x</volume-id>
+      </iso>
+    </media>
+    <tree arch="aarch64">
+      <treeinfo>
+        <family>Red Hat Enterprise Linux</family>
+        <version>^9.7(.([0-9]|[1-9][0-9]))?$</version>
+        <arch>aarch64</arch>
+      </treeinfo>
+    </tree>
+    <tree arch="ppc64le">
+      <treeinfo>
+        <family>Red Hat Enterprise Linux</family>
+        <version>^9.7(.([0-9]|[1-9][0-9]))?$</version>
+        <arch>ppc64le</arch>
+      </treeinfo>
+    </tree>
+    <tree arch="s390x">
+      <treeinfo>
+        <family>Red Hat Enterprise Linux</family>
+        <version>^9.7(.([0-9]|[1-9][0-9]))?$</version>
+        <arch>s390x</arch>
+      </treeinfo>
+    </tree>
+    <tree arch="x86_64">
+      <treeinfo>
+        <family>Red Hat Enterprise Linux</family>
+        <version>^9.7(.([0-9]|[1-9][0-9]))?$</version>
+        <arch>x86_64</arch>
+      </treeinfo>
+    </tree>
+    <resources arch="ppc64le">
+      <minimum>
+        <n-cpus>1</n-cpus>
+        <cpu>1000000000</cpu>
+        <ram>2147483648</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <cpu>4000000000</cpu>
+        <ram>2147483648</ram>
+        <storage>21474836480</storage>
+      </recommended>
+      <maximum>
+        <n-cpus>768</n-cpus>
+        <ram>35184372088832</ram>
+      </maximum>
+    </resources>
+    <resources arch="x86_64">
+      <minimum>
+        <cpu>1000000000</cpu>
+        <ram>1610612736</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <cpu>1000000000</cpu>
+        <ram>1610612736</ram>
+        <storage>21474836480</storage>
+      </recommended>
+      <maximum>
+        <n-cpus>768</n-cpus>
+        <ram>26388279066624</ram>
+      </maximum>
+      <network-install>
+        <ram>3221225472</ram>
+      </network-install>
+    </resources>
+    <resources arch="s390x">
+      <minimum>
+        <ram>1073741824</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <storage>21474836480</storage>
+      </recommended>
+      <maximum>
+        <n-cpus>340</n-cpus>
+        <ram>17592186044416</ram>
+      </maximum>
+      <network-install>
+        <ram>2147483648</ram>
+      </network-install>
+    </resources>
+    <resources arch="aarch64">
+      <minimum>
+        <ram>2147483648</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <storage>21474836480</storage>
+      </recommended>
+      <maximum>
+        <n-cpus>256</n-cpus>
+        <ram>1649267441664</ram>
+      </maximum>
+    </resources>
+    <resources arch="all">
+      <minimum>
+        <n-cpus>1</n-cpus>
+        <cpu>1000000000</cpu>
+        <ram>1610612736</ram>
+        <storage>10737418240</storage>
+      </minimum>
+      <recommended>
+        <cpu>1000000000</cpu>
+        <ram>1610612736</ram>
+        <storage>21474836480</storage>
+      </recommended>
+    </resources>
+    <installer>
+      <script id="http://redhat.com/rhel/kickstart/jeos"/>
+      <script id="http://redhat.com/rhel/kickstart/desktop"/>
+    </installer>
+  </os>
+</libosinfo>
-- 
2.49.0

